app/src/main/java/ar/com/wolox/android/example/ui/auth/login/AuthPresenter.kt

```diff
@@ -1,9 +1,25 @@
 package ar.com.wolox.android.example.ui.auth.login
 
 import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import ar.com.wolox.wolmo.core.util.ToastFactory
 import javax.inject.Inject
 
-class AuthPresenter @Inject constructor() : BasePresenter<AuthView>() {
+class AuthPresenter @Inject constructor(private val toastFactory: ToastFactory) : BasePresenter<AuthView>() {
 
-    fun onLoginButtonClicked() {}
+    fun onLoginButtonClicked(email: String, password: String) {
+        if (password.isEmpty()) view?.setPasswordError("This field is required")
```

Comment:
The presenter should not pass a text directly to the fragment, it's the fragment's resposibility to know what text to show. I suggest to create an enum to store every possible error, and return a list of that enum to the fragment.